crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -379,6 +425,17 @@ module sui::staking_pool {
         staking_pool.pending_total_sui_withdraw
     }
 
+    // Returns true if the provided staking pool is preactive at the provided epoch.
+    public fun is_preactive_at_epoch(pool: &StakingPool, epoch: u64): bool{
```

Comment:
This function only seems to be used in `pool_token_exchange_rate_at_epoch`. 

I'm wondering:
1. Could this function be made non-public? Or, 
2. If it can be removed entirely, and have `pool_token_exchange_rate_at_epoch` default to `initial_exchange_rate` if no table entry for the epoch is found.

Reply:
I'm making this function non-public then.

> have pool_token_exchange_rate_at_epoch default to initial_exchange_rate if no table entry for the epoch is found.

Unfortunately, we can have no table entry for an epoch even when the pool is active, since there might be epochs where we skip reward distribution and the updating of exchange rates. In this case, we need to find the latest entry in the table that's earlier than the given epoch. We have a task on our task tracker to handle this properly.